pkg/kademlia/kademlia.go

```diff
@@ -120,7 +124,119 @@ func (k *Kademlia) GetRoutingTable(ctx context.Context) (dht.RoutingTable, error
 // Bootstrap contacts one of a set of pre defined trusted nodes on the network and
 // begins populating the local Kademlia node
 func (k *Kademlia) Bootstrap(ctx context.Context) error {
-	return k.dht.Bootstrap()
+	if len(k.bootstrapNodes) == 0 {
+		return BootstrapErr.New("no bootstrap nodes provided")
+	}
+	// psuedo randomly select a node from the slice of bootstrap nodes
+	// TODO(coyle): should we instead bootstrap against all the nodes ?
+	bn := k.bootstrapNodes[rand.Intn(len(k.bootstrapNodes))]
+
+	nodes, err := k.nodeClient.Lookup(ctx, bn, k.routingTable.Local())
```

Comment:
instead of calling lookup on the node directly, probably better to look up your own id with k.lookup after bn is in the routing table